app/src/main/java/com/babylon/wallet/android/presentation/transfer/accounts/AccountsChooserDelegate.kt

```diff
@@ -90,30 +95,50 @@ class AccountsChooserDelegate @Inject constructor(
 
         if (!sheetState.isChooseButtonEnabled) return
 
-        _state.update { state ->
-            val ownedAccount = sheetState.ownedAccounts.find { it.address == sheetState.selectedAccount.address }
-            val selectedAccount = if (ownedAccount != null) {
-                TargetAccount.Owned(
-                    account = ownedAccount,
-                    id = sheetState.selectedAccount.id,
-                    assets = sheetState.selectedAccount.assets
+        _state.update {
+            it.copy(
+                sheet = sheetState.copy(
+                    isLoadingAssetsForAccount = true
                 )
-            } else {
-                sheetState.selectedAccount
-            }
+            )
+        }
 
-            val targetAccounts = state.targetAccounts.map { targetAccount ->
-                if (targetAccount.id == selectedAccount.id) {
-                    selectedAccount
+        viewModelScope.launch {
+            _state.update { state ->
+                val ownedAccount = sheetState.ownedAccounts.find { it.address == sheetState.selectedAccount.address }
+                val selectedAccount = if (ownedAccount != null) {
+                    TargetAccount.Owned(
+                        account = ownedAccount,
+                        accountAssetsAddresses = getWalletAssetsUseCase(
```

Comment:
I was thinking, if we could avoid such call when the rule of the target account is not accept known and just return an empty list.

Reply:
@micbakos-rdx check [this](https://github.com/radixdlt/babylon-wallet-android/pull/687/commits/77748613154112ecb9fc0234a2c32f5486323e9d)